src/main/java/org/mitre/fhir/authorization/AuthorizationController.java

```diff
@@ -116,42 +116,48 @@ public String validateClientId(@PathVariable String clientId, HttpServletRequest
 
     if (code != null) {
       fullCodeString = code;
-    } else if (refreshToken != null) {
-      fullCodeString = refreshToken;
-    } else {
-      throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "Invalid code");
-    }
 
-    // the provided code is actualcode.scopes
-    String[] fullCode = fullCodeString.split("\\.");
 
-    String actualCodeOrRefreshToken = fullCode[0];
+      // the provided code is actualcode.scopes
+      String[] fullCode = fullCodeString.split("\\.");
 
-    // if scope was included
-    if (fullCode.length >= 2) {
-      String encodedScopes = fullCode[1];
-      scopes = new String(Base64.getDecoder().decode(encodedScopes));
-    }
+      String actualCodeOrRefreshToken = fullCode[0];
 
-    if (fullCode.length >= 3) {
-      String encodedPatientId = fullCode[2];
-      patientId = new String(Base64.getDecoder().decode(encodedPatientId));
-    }
+      // if scope was included
+      if (fullCode.length >= 2) {
+        String encodedScopes = fullCode[1];
+        scopes = new String(Base64.getDecoder().decode(encodedScopes));
+      }
 
-    if ((code != null && FhirReferenceServerUtils.SAMPLE_CODE.equals(actualCodeOrRefreshToken))) {
-      return generateBearerTokenResponse(request, clientId, scopes, patientId);
-    }
+      if (fullCode.length >= 3) {
+        String encodedPatientId = fullCode[2];
+        patientId = new String(Base64.getDecoder().decode(encodedPatientId));
+      }
 
-    try {
-      if (refreshToken != null
-          && TokenManager.getInstance().authenticateRefreshToken(actualCodeOrRefreshToken)) {
+      if ((code != null && FhirReferenceServerUtils.SAMPLE_CODE.equals(actualCodeOrRefreshToken))) {
         return generateBearerTokenResponse(request, clientId, scopes, patientId);
       }
-    } catch (TokenNotFoundException tokenNotFoundException) {
-      throw new ResponseStatusException(HttpStatus.BAD_REQUEST,
-          "Refresh Token " + refreshToken + " was not found");
+
+    } else if (refreshTokenValue != null) {
+
+      try {
+        if (refreshTokenValue != null
+            && TokenManager.getInstance().authenticateRefreshToken(refreshTokenValue)) {
```

Comment:
```suggestion
        if (TokenManager.getInstance().authenticateRefreshToken(refreshTokenValue)) {
```
The condition refreshTokenValue !=null is always true here because of the encapsulating `if`